### PR TITLE
Add faster approach to clearing docs notebooks with git filter

### PR DESCRIPTION
**Pull request by shahinrostami**
Thank you so much for contributing to Datapane, please help us by filing out this PR template and ensure you have read the [CONTRIBUTING.md](../CONTRIBUTING.md).

It may take a few days to review your PR and merge it - please bear with us!

Thanks!

The git filter runs often (diffs, status), and nbconvert is slow. 

## Prerequsites

- [x] Has been tested locally
- [ ] If a bugfix, have included a breaking test if possible

## Proposed Changes

- Using a custom `jq` solution to clearing our notebooks instead of `nbconvert`, which was very slow.
- Move the commands into a script for better tracking and updating.

### CODE REVIEW

File: docs/clear-notebook-output.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+
+# pass the stdin directly to jq (<&0)
+exec jq --indent 1 \
+  "(.cells[] | select(has(\"outputs\")) | .outputs) = [] | (.cells[] | select(has(\"execution_count\")) | .execution_count) = null | .metadata = {\"language_info\": {\"name\": \"python\"}} | .cells[].metadata |= if has(\"tags\") then {tags} else {} end" \
+  <&0
```
Comment (antdking) [inline docs/clear-notebook-output.sh:9]: We can utilise `HEREDOC`, easier to read and lets us document a more foreign language.

```suggestion
exec jq --indent 1 "$(cat <<'EOF'
    # clear the outputs, if they're set
    (.cells[] | select(has("outputs")) | .outputs) = []
    # clear execution_count, if it's set
    | (.cells[] | select(has("execution_count")) | .execution_count) = null
    # unconditionally set the notebook metadata to be static
    | .metadata = {"language_info": {"name": "python"}}
    # allow tags, nothing else
    | .cells[].metadata |= if has("tags") then {tags} else {} end
EOF
)" <&0
```

File: docs/Taskfile.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ tasks:
     desc: "Install dependencies ready for development on the codebase"
     run: once
     cmds:
-      - cmd: git config filter.clear-notebook-output.clean "cd docs/; poetry run jupyter nbconvert --ClearOutputPreprocessor.enabled=True --ClearMetadataPreprocessor.enabled=True --ClearMetadataPreprocessor.preserve_cell_metadata_mask tags --to=notebook --stdin --stdout --log-level=ERROR"
+      - cmd: git config filter.clear-notebook-output.clean "cd docs/; ./clear-notebook-output.sh"
```
Comment (antdking) [inline docs/Taskfile.yml:12]:
```suggestion
      - cmd: git config filter.clear-notebook-output.clean "./docs/clear-notebook-output.sh"
```

File: docs/README.md [hunk 3]
```diff
@@ -60,7 +60,7 @@ The docs include many Jupyter notebook (`ipynb`) files that must have their outp
 The `.gitattributes` in the `/docs` directory expects the `clear-notebook-output` filter. Please add it to your config with the following command:
 
 ```bash
-git config filter.clear-notebook-output.clean "cd docs/; poetry run jupyter nbconvert --ClearOutputPreprocessor.enabled=True --ClearMetadataPreprocessor.enabled=True --ClearMetadataPreprocessor.preserve_cell_metadata_mask tags --to=notebook --stdin --stdout --log-level=ERROR"
+git config filter.clear-notebook-output.clean "cd docs/; ./clear-notebook-output.sh"
```
Comment (antdking) [inline docs/README.md:63]:
```suggestion
git config filter.clear-notebook-output.clean "./docs/clear-notebook-output.sh"
```

File: docs/clear-notebook-output.sh (hunk 1 above)
Comment (shahinrostami) [reply docs/clear-notebook-output.sh]: Excellent!

**Review by antdking — APPROVED** [review]
nice, much faster. Doesn't impact autocmplete nearly as much

**Pull request merged**